### PR TITLE
python312Packages.oryx: init at 0.2.7

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Add [oryx](https://github.com/jax-ml/oryx), a library for probabilistic programming and deep learning built on top of Jax. 

- **python312Packages.inference-gym: init at 0.0.4**
- **python312Packages.oryx: init at 0.2.7**

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 345450`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python311Packages.inference-gym</li>
    <li>python311Packages.inference-gym.dist</li>
    <li>python311Packages.oryx</li>
    <li>python311Packages.oryx.dist</li>
    <li>python312Packages.inference-gym</li>
    <li>python312Packages.inference-gym.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python311Packages.inference-gym</li>
    <li>python311Packages.inference-gym.dist</li>
    <li>python311Packages.oryx</li>
    <li>python311Packages.oryx.dist</li>
    <li>python312Packages.inference-gym</li>
    <li>python312Packages.inference-gym.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.oryx</li>
    <li>python311Packages.oryx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.inference-gym</li>
    <li>python311Packages.inference-gym.dist</li>
    <li>python312Packages.inference-gym</li>
    <li>python312Packages.inference-gym.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.oryx</li>
    <li>python311Packages.oryx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.inference-gym</li>
    <li>python311Packages.inference-gym.dist</li>
    <li>python312Packages.inference-gym</li>
    <li>python312Packages.inference-gym.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/inference-gym/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  pythonImportsCheck = [ "inference_gym" ];
+
+  # The package does not ship any test.
+  doCheck = false;
+
+  meta = {
+    description = "Oryx is a library for probabilistic programming and deep learning built on top of Jax";
+    homepage = "https://github.com/jax-ml/oryx";
+    changelog = "https://github.com/jax-ml/oryx/releases/tag/v${version}";
```
Comment (natsukium) [inline pkgs/development/python-modules/inference-gym/default.nix:28]:
```suggestion
    description = "Place to exercise inference methods to help make them faster, leaner and more robust";
    homepage = "https://github.com/tensorflow/probability/tree/main/spinoffs/inference_gym";
```

File: pkgs/development/python-modules/oryx/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  pythonImportsCheck = [ "oryx" ];
+
+  nativeCheckInputs = [
+    inference-gym
+    pytestCheckHook
+  ];
+
+  meta = {
+    description = "Oryx is a library for probabilistic programming and deep learning built on top of Jax";
```
Comment (natsukium) [inline pkgs/development/python-modules/oryx/default.nix:47]:
```suggestion
    description = "Library for probabilistic programming and deep learning built on top of Jax";
```

File: pkgs/development/python-modules/inference-gym/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/inference-gym/default.nix]: Oopsie, fixed.

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**